lib/src/combined_wrappers.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// Returns a new list that represents [lists] flattened into a single list.
+///
+/// All methods and accessors treat the new list as-if it were a single
+/// concatenated list, but the underlying implementation is based on lazily
+/// accessing individual list instances.
+///
+/// The returned list is unmodifiable.
+List/*<T>*/ combineLists/*<T>*/(List<List/*<T>*/> lists) {
```

Comment:
Isn't the fact that it is a `CombinedListView` an implementation detail?

From a maintenance perspective you're adding an interface that others could implement/extend or reference even though there isn't any actual real use for doing so.